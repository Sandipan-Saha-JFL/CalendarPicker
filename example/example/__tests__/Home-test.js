import React from 'react';
import Home from '../Components/Home';
import rederer from 'react-test-renderer';

it('home renders correctly', () => {
  rederer.create(<Home />);
});

// it('percent to value function test', () => {
//   let instance = rederer.create(<Home />).getInstance();
//   console.log(instance);
//   expect(instance.percentageToValue(50, 50)).toEqual(25);
// });
// This will not work as functional components does not have instance

it('home snapshot test', () => {
  const snap = rederer.create(<Home />).toJSON();
  expect(snap).toMatchSnapshot();
});

it('home text input identification', () => {
  const elementTree = rederer.create(<Home />).toJSON();
  let element = elementTree.children.find(
    node => node.props.testID === 'sampletext',
  );

  expect(element).toBeDefined();
});
