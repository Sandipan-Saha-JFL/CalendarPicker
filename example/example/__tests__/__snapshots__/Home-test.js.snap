// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home snapshot test 1`] = `
<View>
  <Text>
    Home component view test percentage 
    50
    % with
     
  </Text>
  <TextInput
    onChangeText={[Function]}
    testID="sampletext"
  />
  <Text>
    This is profile
  </Text>
</View>
`;
